shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/ManagedBridgeSyncServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+package com.redhat.service.smartevents.shard.operator.v2;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.smartevents.infra.v2.api.models.ConditionStatus;
+import com.redhat.service.smartevents.infra.v2.api.models.OperationType;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.BridgeDTO;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.BridgeStatusDTO;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.ConditionDTO;
+import com.redhat.service.smartevents.shard.operator.v2.converters.BridgeStatusConverter;
+import com.redhat.service.smartevents.shard.operator.v2.resources.ManagedBridge;
+
+import static com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions.DP_AUTHORISATION_POLICY_READY_NAME;
+import static com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions.DP_CONFIG_MAP_READY_NAME;
+import static com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions.DP_KNATIVE_BROKER_READY_NAME;
+import static com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions.DP_NETWORK_RESOURCE_READY_NAME;
+import static com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions.DP_SECRET_READY_NAME;
+
+@ApplicationScoped
+public class ManagedBridgeSyncServiceImpl implements ManagedBridgeSyncService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ManagedBridgeSyncServiceImpl.class);
+
+    @Inject
+    ManagerClient managerClient;
+
+    @Inject
+    ManagedBridgeService managedBridgeService;
+
+    @Override
+    public void syncManagedBridgeWithManager() {
+        managerClient.fetchBridgesToDeployOrDelete()
+                .onItem()
+                .invoke(this::processDelta)
+                .subscribe().with(
+                        success -> LOGGER.debug("Successfully processed ManagedBridge deltas"),
+                        error -> LOGGER.error("Failed to process ManagedBridge deltas", error));
+    }
+
+    @Override
+    public void syncManagedBridgeStatusBackToManager() {
+        managerClient.fetchBridgesToDeployOrDelete()
```

Comment:
Why does it fetch the resources from the manager?
I think we can even do a simple "send all statuses" reading all resources available in kubernetes. In this way we should avoid inconsistencies.
For example what if a resource switches to failed? This resource is not in the list of "toDeployOrDelete".

Reply:
I agree with @danielezonca.

IMO k8s should be queried for the resources' status conditions.

The `operator` is the "source of truth" of `Condition`'s during provisioning/deprovisioning; not the `manager`.

Furthermore the `manager` holds a [temporary](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/468203f50e91160441a47cda31801438e84d7787/infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/DefaultConditions.java#L6) `Condition` as a place-holder until the first update from the `operator`. Then it overwrites the temporary one with the set from the `operator`. We took this approach so that the `manager` could be blissfully unaware of what `Condition`'s are required to provision/deprovision a resource. If, the set of `Condition`'s changes the `manager` works without change.

Reply:
I am fetching the bridges from manager so that we could also send back status for those resources which are deleted. 
If I just send back status of only those resources whose CRD exists then status for deleted resources would never sent.

@danielezonca @manstis I would remove this logic if you guys think it would not required for now.

Reply:
> I am fetching the bridges from manager so that we could also send back status for those resources which are deleted.
> If I just send back status of only those resources whose CRD exists then status for deleted resources would never sent.

Good point, but at the same time the other scenario that I mentioned about a failed resource is not covered by current code.
If we agree that the logic is "send all", the manager should be able to do the delta: if a resource in deleting state is not included in the returned list, it means that it is deleted.

We can even consider a mixed approach where we do both but if we take this statement

> The operator is the "source of truth" of Condition's during provisioning/deprovisioning; not the manager.

it is not wrong to assume that a `deleting` that is not returned by the shard implies that is deleted

Reply:
@danielezonca Can you clarify what you mean when you say:

```
For example what if a resource switches to failed? This resource is not in the list of "toDeployOrDelete".
```
At the minute the Control Plane will provide all Bridges/Processors when the conditions added by the Control Plane are `true` e.g the Control Plane has finished the provisioning that it is responsible for e.g DNS, Kafka Topic. It's then up to the data plane to do it's provisioning which will either succeed or not.

Reply:
@robpblake 
I might be wrong but as far as I understand the `fetchBridgesToDeployOrDelete` method is not going to return a resource that has been fully provisioned and now it is just running.
If that assumption is correct, I'm not sure if it is possible to manage a scenario where a resource that is currently running well, move to a failed status (for whatever reason, i.e. a broken update) because this resource is not in the list returned by the manager.

The second quote is just an alternative proposal but I agree that it is better (if possible) to have an explicit `deleted=true`

Reply:
@danielezonca 

> I might be wrong but as far as I understand the fetchBridgesToDeployOrDelete method is not going to return a resource that has been fully provisioned and now it is just running.

That understanding is wrong.

This is something @r00ta and I argued until we were blue in the face; but it was eventually conceded that the `manager` will furnish the `operator` with a list of ALL resources that the `manager` has finished preliminary work for. This includes all new resources, all updated resources, all provisioned resources, all failed resources. EVERYTHING.

Reply:
Ok if so my comment doesn't apply and I only suggest to rename the method `fetchBridgesToDeployOrDelete` because it sounds to me a bit misleading :) (I think even a simple `fetchBridgesForDataPlane` or similar can be enough)
👍